### PR TITLE
refactor(sql): move dialects to always-importable location

### PR DESCRIPTION
**Pull request by cpcloud**
This PR consolidates our dialects into a single file (`ibis/backends/base/sqlglot/dialects`) to allow reuse without implicitly needing to import drivers or other backend-specific dependencies.

### CODE REVIEW

**Review by jcrist — APPROVED** [review]
Nice

File: ibis/backends/base/sqlglot/dialects.py [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+# TODO: can delete this after bumping sqlglot version > 20.9.0
+Oracle.Generator.TYPE_MAPPING |= {
+    sge.DataType.Type.TIMETZ: "TIME",
+    sge.DataType.Type.TIMESTAMPTZ: "TIMESTAMP",
+}
+Oracle.Generator.TZ_TO_WITH_TIME_ZONE = True
+
+
+class Polars(Postgres):
+    """Subclass of Spark dialect for PySpark.
```
Comment (cpcloud) [inline ibis/backends/base/sqlglot/dialects.py:168]: Definitely not correct :)

Comment (cpcloud) [reply ibis/backends/base/sqlglot/dialects.py]: Fixed.

**Comment by cpcloud** [comment]
Gah, I keep forgetting we don't have CI blocking on the-epic-split. Enabling that now.

**Pull request merged**